pfcpiface/conn.go

```diff
@@ -82,52 +93,65 @@ func pfcpifaceMainLoop(upf *upf, accessIP, coreIP, sourceIP, smfName string) {
 		go pconn.manageSmfConnection(sourceIP, accessIP, smfName, conn, cpConnectionStatus)
 	}
 
-	// Initialize pkt buf
-	buf := make([]byte, PktBufSz)
+	pfcpRcvdPktsChan := make(chan *rcvdPacket,1000)
+	pfcpParsedPktsChan := make(chan *parsedPacket,1000)
 	// Initialize pkt header
 
-	for {
-		err := conn.SetReadDeadline(time.Now().Add(readTimeout))
-		if err != nil {
-			log.Fatalln("Unable to set deadline for read:", err)
-		}
-		// blocking read
-		n, addr, err := conn.ReadFrom(buf)
-		if err != nil {
-			if err, ok := err.(net.Error); ok && err.Timeout() {
-				// do nothing for the time being
-				log.Println(err)
-				cpConnected = false
-				if manageConnection {
-					cpConnectionStatus <- cpConnected
+	pfcpPacketReader := func(pfcpRcvdPkts chan *rcvdPacket) {
+		for {
+			err := conn.SetReadDeadline(time.Now().Add(readTimeout))
+			if err != nil {
+				log.Fatalln("Unable to set deadline for read:", err)
+			}
+			pkt := new(rcvdPacket)
+			// blocking read
+			n, addr, err := conn.ReadFrom(pkt.Buf[:1500])
+			if err != nil {
+				if err, ok := err.(net.Error); ok && err.Timeout() {
+					// do nothing for the time being
+					log.Println(err)
+					cpConnected = false
+					if manageConnection {
+						cpConnectionStatus <- cpConnected
+					}
+					cleanupSessions()
```

Comment:
when connection breaks along with cleaning up sessions we should clear any packets in the new channels

Reply:
Yeah, I realized that cleanup needs better placement and better handling in general.

Reply:
break it up in to separate function. same as comment below.